### PR TITLE
commit 1

### PR DESCRIPTION
**Pull request by dheerajsrinath**
Verify if email id is present

### CODE REVIEW

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 1]
```diff
@@ -38,4 +38,28 @@ public void testLoginValidation(){
         softAssertions.assertThat(responseElement.getPayload().getUid()).as("Uid not same as body")
                 .isEqualTo(email);
     }
+
+    @Test
+    public void testLoginAuthentication(){
+
+        com.training.pojos.ga.authentication.Request request = new com.training.pojos.ga.authentication.Request();
+        request.setEmail("email@email.com");
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java]: define it globally

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 2]
```diff
@@ -38,4 +38,28 @@ public void testLoginValidation(){
...
+
+        response = new RestEngine().getResponse("/en/al/web/v3/guestAccounts/validation",headerMap,
+                new Gson().toJson(request));
+
+        com.training.pojos.ga.authentication.Response responseElement =
+                response.as(com.training.pojos.ga.authentication.Response.class);
+
+        Assertions.assertEquals(200,responseElement.getStatus(),"Request Unsuccessful");
+
+        softAssertions.assertThat(responseElement.getErrors().length).as("Error present in response")
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java]: define it globally

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 3]
```diff
@@ -38,4 +38,28 @@ public void testLoginValidation(){
...
+        com.training.pojos.ga.authentication.Request request = new com.training.pojos.ga.authentication.Request();
+        request.setEmail("email@email.com");
+
+        response = new RestEngine().getResponse("/en/al/web/v3/guestAccounts/validation",headerMap,
+                new Gson().toJson(request));
+
+        com.training.pojos.ga.authentication.Response responseElement =
+                response.as(com.training.pojos.ga.authentication.Response.class);
+
+        Assertions.assertEquals(200,responseElement.getStatus(),"Request Unsuccessful");
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java]: define it globally

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 4]
```diff
@@ -38,4 +38,28 @@ public void testLoginValidation(){
...
+
+        softAssertions.assertThat(responseElement.getErrors().length).as("Error present in response")
+                .isEqualTo(0);
+
+        softAssertions.assertThat(responseElement.getPayload().getAccountStatus()).as("Account Status mismatch")
+                .isEqualTo("EXISTS");
+
+        softAssertions.assertThat(responseElement.getPayload().isUid()).as("Email doesn't exists")
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java:62]: responseElement.getPayload() = more than  two times, define it  in a varible and use inside assert

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 5]
```diff
@@ -10,7 +11,9 @@
 import org.junit.jupiter.api.Test;
 
 public class GuestAccountTests extends ApiBaseTest {
-
+    String email = "email@email.com";
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java]: should be final

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 6]
```diff
@@ -10,6 +11,9 @@
 import org.junit.jupiter.api.Test;
 
 public class GuestAccountTests extends ApiBaseTest {
+    private final String email = "email@email.com";
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java]: should be capital

File: src/test/java/com/training/api/tests/ga/GuestAccountTests.java [hunk 7]
```diff
@@ -20,22 +24,46 @@ public void startup(){
...
+
+        softAssertions.assertThat(responseElement.getErrors().length).as(ERROR_PRESENT)
+                .isEqualTo(0);
+
+        Payload payload = responseElement.getPayload();
+        softAssertions.assertThat(payload.getAccountStatus()).as("Account Status mismatch")
+                .isEqualTo("EXISTS");
+
+        softAssertions.assertThat(payload.isUid()).as("Email doesn't exists")
+                .isEqualTo(true);
```
Comment (akhil7000) [inline src/test/java/com/training/api/tests/ga/GuestAccountTests.java]: lets use assertTrue

**Review by akhil7000 — CHANGES_REQUESTED** [review]

**Review by akhil7000 — APPROVED** [review]

**Pull request merged**